lib/charms/lte_core_interface/v0/lte_core_interface.py

```diff
@@ -196,9 +198,8 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
             return
         remote_app_relation_data = relation.data[relation.app]
         if not self._relation_data_is_valid(dict(remote_app_relation_data)):
-            logger.warning(
-                "Provider relation data did no pass JSON Schema validation: \n%s",
-                event.relation.app[event.app],
+            logger.error(
+                "Provider relation data did not pass JSON schema validation."  # noqa: E501
```

Comment:
Do we need a `_relation_data_is_valid` method to validate this bit?
It seems as a bit of duplication for me.

Reply:
With `_relation_data_is_valid` we validate the `jsonschema`, `__mme_ipv4_address_is_valid` validates the ip address itself, as the jsonchema `format` field is just informative, so It doesn't perform any check. Or do you mean that validating both is not worth it? Let me know.

Reply:
I just meant that `_relation_data_is_valid` calls the `validate` function and returns True or False, and then we check here again.
couldn't we just call the `validate` function here?

Reply:
My rationale here is just to make it a bit more readable maintaining the guard condition pattern:

```python
def blabla():
# Guard conditions
if not something:
    logger.error(something)
    return
if not something2:
   logger.error(something2)
   return

# actual method code
```

In the other case would be something like
```python
def blabla():
if not something:
    logger.error(something)
    return

try:
    # some code
except:
    logger.error("something about the exception")
    return

# actual method code
```

I prefer the first one, but It might be a bit overkill, Idk, let me know, I don't have a strong preference